library/GitHub/Workflow/Command.hs

```diff
@@ -0,0 +1,70 @@
+module GitHub.Workflow.Command
+  ( -- * The Command type
+    Command
+
+    -- * Ancillary types
+  , Name
+  , Properties
+  , Key
+  , Value
+  , Message
+
+    -- * Building commands
+  , command
+  , message
+  , property
+
+    -- * Printing commands
+  , toByteString
+  , toByteStringBuilder
+  ) where
+
+import Control.Category
+import Data.ByteString
+import Data.ByteString.Builder qualified as BSB
+import Data.ByteString.Lazy qualified as BSL
+import Data.Foldable qualified as Foldable
+import Data.Maybe (Maybe)
+import Data.String (IsString (fromString))
+import GitHub.Workflow.Command.Message (Message)
+import GitHub.Workflow.Command.Message qualified as Message
+import GitHub.Workflow.Command.Name (Name)
+import GitHub.Workflow.Command.Name qualified as Name
+import GitHub.Workflow.Command.Properties (Key, Properties, ToValue, Value)
+import GitHub.Workflow.Command.Properties qualified as Properties
+import Prelude (Eq, Ord, Show, (<>))
+
+data Command = Command
+  { name :: Name
+  , properties :: Properties
+  , message :: Message
+  }
+  deriving stock (Eq, Ord, Show)
+
+instance IsString Command where
+  fromString = fromString @Name >>> command
```

Comment:
The use of `>>>` seems a bit gratuitous to me.

```suggestion
  fromString = command . fromString
```

(I chose to drop the type application too, though I'm not strongly against it. It just doesn't seem like a leap worth annotating to know what type `command` takes.)